src/main/scala/firrtl/passes/DeadCodeElimination.scala

```diff
@@ -34,22 +34,22 @@ object DeadCodeElimination extends Pass {
       out
     }
 
-    def maybeEliminate(x: Statement, name: String) =
-      if (referenced(name)) x
+    def maybeEliminate(x: Statement, name: String, exception: String): Statement =
+      if (referenced(name) && name != exception) x
```

Comment:
Why is this necessary? The effectively neuters the DefRegister maybeEliminate. It now never happens.